crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -479,3 +499,99 @@ impl<F: Send + Clone> ValidateRequest<F, api::PaymentsRequest> for PaymentConfir
+    let order_details_outside_metadata_req = request.clone().order_details;
+    let order_details_metadata_req = request.clone().metadata.and_then(|meta| meta.order_details);
+
+    if order_details_outside_metadata_req
+        .clone()
+        .zip(order_details_metadata_req.clone())
+        .is_some()
+    {
+        Err(errors::ApiErrorResponse::NotSupported { message: "order_details cannot be present both inside and outside metadata in payments request".to_string() })?
+    }
+    if order_details_metadata_db
+        .clone()
+        .zip(order_details_outside_metadata_db.clone())
+        .is_some()
+    {
+        Err(errors::ApiErrorResponse::NotSupported { message: "order_details cannot be present both inside and outside metadata in payment intent in db".to_string() })?
+    }
+    let order_details_outside = match order_details_outside_metadata_req {
+        Some(order) => match order_details_metadata_db {
+            Some(_) => Err(errors::ApiErrorResponse::NotSupported {
+                message: "order_details previously present inside of metadata".to_string(),
+            })?,
+            None => Some(order),
+        },
+        None => match order_details_metadata_req {
+            Some(_order) => match order_details_outside_metadata_db {
+                Some(_) => Err(errors::ApiErrorResponse::NotSupported {
+                    message: "order_details previously present outside of metadata".to_string(),
+                })?,
+                None => None,
+            },
+            None => None,
+        },
+    };
+    let metadata_with_order_details = request.clone().metadata.map(|meta| {
+        let transformed_metadata = match parsed_metadata_db {
+            Some(meta_db) => api_models::payments::Metadata {
+                order_details: meta.order_details,
+                ..meta_db
+            },
+            None => meta,
+        };
+        let transformed_metadata_value =
+            Encode::<api_models::payments::Metadata>::encode_to_value(&transformed_metadata)
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Encoding Metadata to value failed")
+                .unwrap_or_default();
+        Secret::new(transformed_metadata_value)
+    });
+    if order_details_outside.is_some() {
+        payment_intent.order_details = order_details_outside.map(|o| {
+            o.iter()
+                .map(|order| {
+                    let encoded_order = <api_models::payments::OrderDetailsWithAmount as Encode<
+                        '_,
+                        frunk_core::labelled::chars::P,
+                    >>::encode_to_value(order)
+                    .change_context(errors::ApiErrorResponse::NotSupported {
+                        message: "failed while encoding order_details".to_string(),
+                    })
+                    .unwrap_or_default();
```

Comment:
If any errors occurs, we need to pass default?

If so why we need to change_context?

If error occurs, how we know about that error.